### PR TITLE
Feat: Adds `Link` component

### PR DESCRIPTION
**Pull request by ArthurTriis1**
## What's the purpose of this pull request?

- Moves `Link` Component from @faststore/ui to @faststore/components
- Adds `Link` styles
- Updates `Link` documentation page on docusaurus

## How to test it?

run yarn start in app/docs folder and checks the Icon component or test it using the doc [preview link.](https://faststore-git-feat-replace-fs-ui-link-config-ui-faststore.vercel.app/reference/ui/atoms/Link)

### Starters Deploy Preview

https://github.com/vtex-sites/nextjs.store/pull/317

### CODE REVIEW

File: apps/docs/docs/reference/ui/atoms/Link.mdx [hunk 1]
```diff
@@ -1,21 +1,44 @@
...
+  - react
 sidebar_custom_props: 
   image: https://vtexhelp.vtexassets.com/assets/docs/src/Link___ae95650d3842207e2d790446c3418086.png
 ---
 
 import PropsSection from "@site/src/components/PropsSection/PropsSection";
+import { TokenTable, TokenRow, TokenDivider } from '@site/src/components/Tokens'
 
-# Link
+Link performs the same behavior of a regular anchor tag (`a`), but can be used as any HTML tag and component, including spacifics Frameworks link components.
```
Comment (eduardoformiga) [inline apps/docs/docs/reference/ui/atoms/Link.mdx:14]:
```suggestion
Link performs the same behavior of a regular anchor tag (`a`), but can be used as any HTML tag and component, including specific Frameworks link components.
```

File: apps/docs/docs/reference/ui/atoms/Link.mdx [hunk 2]
```diff
@@ -1,21 +1,44 @@
...
 
-The `Link` component behaves as a regular anchor tag (`<a>`), but can be used as any HTML tag.
+## Overview
+
+```tsx live
+<div style={{ display: 'flex', columnGap: '16px', alignItems: 'center' }}>
```
Comment (eduardoformiga) [inline apps/docs/docs/reference/ui/atoms/Link.mdx:19]: what do you think about using the `overviewSection` class instead?

File: apps/docs/docs/reference/ui/atoms/Link.mdx [hunk 3]
```diff
@@ -1,21 +1,44 @@
...
+  <Link variant="default">Default</Link>
+  <Link variant="display">Display</Link>
+  <Link variant="footer">Footer</Link>
+  <Link variant="inline">Inline</Link>
+  <Link inverse={true}>Inverse</Link>
+</div>
+```
```
Comment (eduardoformiga) [inline apps/docs/docs/reference/ui/atoms/Link.mdx:27]: Should we include this divisor between the sections?
```suggestion

---

```

File: packages/components/src/index.ts [hunk 4]
```diff
@@ -11,6 +11,8 @@ export { default as Label } from './atoms/Label'
 export type { LabelProps } from './atoms/Label'
 export { default as Radio } from './atoms/Radio'
 export type { RadioProps } from './atoms/Radio'
+export { default as Link } from './atoms/Link'
+export type { LinkProps, LinkElementType } from './atoms/Link'
```
Comment (eduardoformiga) [inline packages/components/src/index.ts:15]: ⛏️  please place it after the label for the sake of sorting alphabetically

File: packages/ui/src/components/atoms/Link/Link.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+export const Template = ({ children, ...args }) => (
+  <Link {...args}>{children}</Link>
+)
+
+<header>
+
+# Link
+
+Link performs the same behavior of a regular anchor tag (`a`), but can be used as any HTML tag and component, including spacifics Frameworks link components.
```
Comment (eduardoformiga) [inline packages/ui/src/components/atoms/Link/Link.stories.mdx:48]:
```suggestion
Link performs the same behavior of a regular anchor tag (`a`), but can be used as any HTML tag and component, including specific Frameworks link components.
```

File: packages/ui/src/components/atoms/Link/Link.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,125 @@
+import { Meta, Canvas, Story, ArgsTable } from '@storybook/addon-docs'
```
Comment (eduardoformiga) [inline packages/ui/src/components/atoms/Link/Link.stories.mdx:1]: Maybe I got it wrong but aren't we planning to remove the storybook from the `faststore/ui`?

File: packages/ui/src/styles/components.scss [hunk 7]
```diff
@@ -3,6 +3,7 @@
 @import "../components/atoms/Button/styles";
 @import "../components/atoms/Checkbox/styles";
 @import "../components/atoms/Radio/styles";
+@import "../components/atoms/Link/styles";
```
Comment (eduardoformiga) [inline packages/ui/src/styles/components.scss:6]: ⛏️  could you place it after the checkbox, please?

**Review by eduardoformiga — COMMENTED** [review]
LGTM! 
Left some comments during the review.

can't see the inverse link
I think we can tweak these sections from the docs.
Screen Shot 2022-12-07 at 16 26 12

links together
Screen Shot 2022-12-07 at 16 24 29

File: apps/docs/docs/reference/ui/atoms/Link.mdx (hunk 3 above)
Comment (ArthurTriis1) [reply apps/docs/docs/reference/ui/atoms/Link.mdx:27]: I based on the documentation of the button

File: apps/docs/docs/reference/ui/atoms/Link.mdx [hunk 8]
```diff
@@ -1,35 +1,49 @@
...
+<div className="overviewSection">
+  <Link variant="default">Default</Link>
+  <Link variant="display">Display</Link>
+  <Link variant="inline">Inline</Link>
+  <Link size="small">Small</Link>
+  <Link inverse={true}>Inverse</Link>
+</div>
+```
 
 ## Import
```
Comment (renatamottam) [inline apps/docs/docs/reference/ui/atoms/Link.mdx:28]:
```suggestion

---

## Import
```

File: apps/docs/docs/reference/ui/atoms/Link.mdx [hunk 9]
```diff
@@ -1,35 +1,49 @@
...
-```tsx live
-<Link href="/">
-    A default link
-</Link>
+```tsx
+import '@faststore/ui/src/components/atoms/Link/styles.scss'
 ```
 
-### As `div`
+## Usage
```
Comment (renatamottam) [inline apps/docs/docs/reference/ui/atoms/Link.mdx:42]:
```suggestion
---

## Usage
```

File: apps/docs/docs/reference/ui/atoms/Link.mdx [hunk 10]
```diff
@@ -40,8 +54,120 @@ For example, if someone wants to use a wrapper for any external link component (
 
 Besides those attributes, the following props are also supported:
 
-<PropsSection name="StorybookLink" />
+<PropsSection name="Link" />
+
+## Design Tokens
```
Comment (renatamottam) [inline apps/docs/docs/reference/ui/atoms/Link.mdx:59]:
```suggestion

---

## Design Tokens
```

File: apps/docs/docs/reference/ui/atoms/Link.mdx [hunk 11]
```diff
@@ -40,8 +54,120 @@ For example, if someone wants to use a wrapper for any external link component (
...
+    token="--fs-link-transition-timing"
+    value="var(--fs-transition-timing)"
+  />
+</TokenTable>
+
+---
+
+## Variants
+
+### Inverse
```
Comment (renatamottam) [inline apps/docs/docs/reference/ui/atoms/Link.mdx:99]:
```suggestion

### Variants

#### Inverse

```

File: apps/docs/docs/reference/ui/atoms/Link.mdx [hunk 12]
```diff
@@ -40,8 +54,120 @@ For example, if someone wants to use a wrapper for any external link component (
...
+    value="var(--fs-link-text-color)"
+    globalValue="var(--fs-color-link)"
+    isColor
+  />
+</TokenTable>
+
+## Size
+
+### Small
+
```
Comment (renatamottam) [inline apps/docs/docs/reference/ui/atoms/Link.mdx:162]:
```suggestion

#### Display

```tsx live
<Link variant="display">Display Link</Link>
```

<TokenTable>
  <TokenRow
    token="--fs-link-display-text-line-height"
    value="var(--fs-link-text-line-height)"
  />
  <TokenRow
    token="--fs-link-display-text-color"
    value="var(--fs-color-text-display)"
    isColor
  />
  <TokenRow
    token="--fs-link-display-text-color-visited"
    value="var(--fs-link-display-text-color)"
    globalValue="var(--fs-color-text-display)"
    isColor
  />
</TokenTable>

#### Inline

```tsx live
<Link variant="inline">Inline Link</Link>
```

<TokenTable>
  <TokenRow token="--fs-link-inline-padding" value="0" />
  <TokenRow token="--fs-link-inline-text-decoration" value="underline" />
  <TokenRow
    token="--fs-link-inline-text-color"
    value="var(--fs-link-text-color)"
    globalValue="var(--fs-color-link)"
    isColor
  />
</TokenTable>

### Small

```

**Review by renatamottam — COMMENTED** [review]
Nice one, Arthur!
I left a few doc adjusts, you can check `Input` page as reference.

File: apps/docs/docs/reference/ui/atoms/Link.mdx [hunk 13]
```diff
@@ -1,35 +1,49 @@
...
+  - react
 sidebar_custom_props: 
   image: https://vtexhelp.vtexassets.com/assets/docs/src/Link___ae95650d3842207e2d790446c3418086.png
 ---
 
 import PropsSection from "@site/src/components/PropsSection/PropsSection";
+import { TokenTable, TokenRow, TokenDivider } from '@site/src/components/Tokens'
 
-# Link
+Link performs the same behavior of a regular anchor tag (`a`), but can be used as any HTML tag and component, including spacific Frameworks link components.
```
Comment (hellofanny) [inline apps/docs/docs/reference/ui/atoms/Link.mdx:14]:
```suggestion
Link performs the same behavior of a regular anchor tag (`a`), but can be used as any HTML tag and component, including specific Frameworks link components.
```

File: packages/components/src/atoms/Link/Link.tsx [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+ */
+  variant?: Variant
+  /**
+  * Defines use of inverted color.
+  */
+  inverse?: boolean
+  /**
+   * Defines size os element
+   */
+  size?: 'small' | 'regular'
```
Comment (hellofanny) [inline packages/components/src/atoms/Link/Link.tsx:23]: nice 🆒

File: packages/components/src/atoms/Link/Link.tsx [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+>
+
+type LinkComponent = <C extends LinkElementType>(
+  props: LinkProps<C>
+) => ReactElement | null
+
+const Link: LinkComponent = forwardRef(function Link<
+  C extends LinkElementType = 'a'
+>(
+  { as, children, variant = 'default', size="regular", inverse, testId = 'store-link', ...otherProps }: LinkProps<C>,
```
Comment (hellofanny) [inline packages/components/src/atoms/Link/Link.tsx:40]:
```suggestion
  { as, children, variant = 'default', size="regular", inverse, testId = 'fs-link', ...otherProps }: LinkProps<C>,
```

Comment (hellofanny) [reply packages/components/src/atoms/Link/Link.tsx:40]: I think we renamed everything from `store` for `fs`

File: packages/components/src/atoms/Link/Link.tsx [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+  props: LinkProps<C>
+) => ReactElement | null
+
+const Link: LinkComponent = forwardRef(function Link<
+  C extends LinkElementType = 'a'
+>(
+  { as, children, variant = 'default', size="regular", inverse, testId = 'fs-link', ...otherProps }: LinkProps<C>,
+  ref?: PolymorphicRef<C>
+) {
+  const Component = as ?? 'a'
```
Comment (emersonlaurentino) [inline packages/components/src/atoms/Link/Link.tsx:43]:
```suggestion
  { as: Component = "a", children, variant = 'default', size="regular", inverse, testId = 'fs-link', ...otherProps }: LinkProps<C>,
  ref?: PolymorphicRef<C>
) {
```

Comment (emersonlaurentino) [reply packages/components/src/atoms/Link/Link.tsx:43]: You don't need to create a new variable for it, you can just rename the prop and add an initial value.

**Review by emersonlaurentino — APPROVED** [review]

File: packages/components/src/atoms/Link/Link.tsx (hunk 16 above)
Comment (ArthurTriis1) [reply packages/components/src/atoms/Link/Link.tsx]: I had problems with typing using this approach

File: apps/docs/docs/reference/ui/atoms/Link.mdx [hunk 17]
```diff
@@ -24,24 +51,162 @@ import { Link } from '@faststore/ui'
...
+<PropsComponent
+  name={'testId'}
+  type={'string'}
+  defaultValue={'fs-link'}
+  description={'ID to find this component in testing tools (e.g.: cypress, testing library, and jest).'}
+  required={false}
+/>
+<PropsComponent
+  name={'variant'}
+  type={'"default" | "display" | "footer" | "inline"'}
```
Comment (renatamottam) [inline apps/docs/docs/reference/ui/atoms/Link.mdx:70]:
```suggestion
  type={'"default" | "display" | "inline"'}
```

File: apps/docs/docs/reference/ui/atoms/Link.mdx [hunk 18]
```diff
@@ -24,24 +51,162 @@ import { Link } from '@faststore/ui'
...
+  />
+</TokenTable>
+
+### Variants
+
+#### Inverse
+
+```tsx live
+<Link inverse={true}>Inverse Link</Link>
+```
```
Comment (renatamottam) [inline apps/docs/docs/reference/ui/atoms/Link.mdx:133]:
```suggestion
```tsx live
<div style={{ backgroundColor: '#171a1c', padding: '1rem' }}>
  <Link inverse={true}>Inverse Link</Link>
</div>
```
```

File: apps/docs/docs/reference/ui/atoms/Link.mdx [hunk 19]
```diff
@@ -24,24 +51,162 @@ import { Link } from '@faststore/ui'
...
+  />
+</TokenTable>
+
+---
 
 ## Customization
 
+For further customization, you can use the following data attributes:
+
 `data-fs-link`
```
Comment (renatamottam) [inline apps/docs/docs/reference/ui/atoms/Link.mdx:212]:
```suggestion
`data-fs-link`

`data-fs-link-variant="default"`

`data-fs-link-variant="display"`

`data-fs-link-variant="inline"`

`data-fs-link-size="small"`

`data-fs-link-inverse="true"`
```

**Review by renatamottam — APPROVED** [review]

File: packages/components/src/atoms/Link/Link.tsx [hunk 20]
```diff
@@ -0,0 +1,59 @@
...
+  * Defines use of inverted color.
+  */
+  inverse?: boolean
+  /**
+   * Defines size os element
+   */
+  size?: 'small' | 'regular'
+}
+
+export type LinkElementType = ElementType
```
Comment (lucasfp13) [inline packages/components/src/atoms/Link/Link.tsx:24]: Do we have a specific case that we gonna need this type?

**Review by lucasfp13 — APPROVED** [review]
💯

File: packages/components/src/atoms/Link/Link.tsx [hunk 21]
```diff
@@ -0,0 +1,59 @@
...
+
+type LinkBaseProps = {
+  /**
+   * ID to find this component in testing tools (e.g.: cypress, testing library, and jest).
+   */
+  testId?: string
+  /**
+ * Specifies the component variant.
+ */
+  variant?: 'default' | 'display' | 'footer' | 'inline'
```
Comment (hellofanny) [inline packages/components/src/atoms/Link/Link.tsx:13]:
```suggestion
  variant?: 'default' | 'display' | 'inline'
```

Comment (hellofanny) [reply packages/components/src/atoms/Link/Link.tsx:13]: haven't we remove this variant? 👁️

File: apps/docs/docs/reference/ui/atoms/Link.mdx [hunk 22]
```diff
@@ -1,21 +1,48 @@
...
-The `Link` component behaves as a regular anchor tag (`<a>`), but can be used as any HTML tag.
+## Overview
+
+```tsx live
+<div className="overviewSection">
+  <Link variant="default">Default</Link>
+  <Link variant="display">Display</Link>
+  <Link variant="inline">Inline</Link>
+  <Link size="small">Small</Link>
+  <Link inverse={true}>Inverse</Link>
```
Comment (hellofanny) [inline apps/docs/docs/reference/ui/atoms/Link.mdx:24]: I couldn't see the inverse variant here ahaha. Maybe we can add a background sample here?

image

Comment (hellofanny) [inline apps/docs/docs/reference/ui/atoms/Link.mdx:24]:
```suggestion
<div style={{ backgroundColor: '#171a1c', padding: '0.5rem' }}>
   <Link inverse={true}>Inverse Link</Link>
 </div>
```

Comment (hellofanny) [reply apps/docs/docs/reference/ui/atoms/Link.mdx:24]: Or we can do the same as we did on storybook: (Adds background color to the whole section)

image

File: packages/components/src/atoms/Link/Link.tsx (hunk 20 above)
Comment (ArthurTriis1) [reply packages/components/src/atoms/Link/Link.tsx:24]: I had problems with different versions of react, this type can fix it

File: packages/components/src/atoms/Link/Link.tsx (hunk 21 above)
Comment (ArthurTriis1) [reply packages/components/src/atoms/Link/Link.tsx:13]: Yes... I think it came back e some rebase

**Pull request merged**